src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -957,12 +957,88 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             }
 
             if (format.Length == 1) {
+                switch (format) {
+                    case "o":
+                    case "O":
+                        // Fast track for round trip format without going through a format string.
+                        return RoundTripFormat(ref dateTime);
+                }
+
                 format = ExpandPredefinedFormat(format, ref dateTime, ref dtfi, ref offset);
-            }            
+            }      
 
             return (FormatCustomized(dateTime, format, dtfi, offset));
         }
-    
+
+        internal static string RoundTripFormat(ref DateTime dateTime)
+        {
+            StringBuilder result = StringBuilderCache.Acquire();
+            Append(result, dateTime.Year, 4);
```

Comment:
Note that calling `.Year`, `.Month`, etc are all doing quite a bit of work, and you are doing this each and every time.
This is computationally expensive, and you can probably avoid doing this if you build the entire thing in one go.